src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -133,7 +137,7 @@ internal RouteEndpointBuilder GetSingleRouteEndpointBuilder()
         // ApplicationBuilder.Build(). This was observed in MapSignalRTests and is not very useful. Maybe if we come up
         // with a better heuristic for what a useful method name is, we could use it for everything. Inline lambdas are
         // compiler generated methods so they are filtered out even for route handlers.
-        if (isRouteHandler && TypeHelper.TryGetNonCompilerGeneratedMethodName(handler.Method, out var methodName))
+        if (isRouteHandler && TypeHelper.TryGetNonCompilerGeneratedMethodName(entry.Method, out var methodName))
```

Comment:
Other pieces of code needs to be updated. I would resolve the method info at the top and use it everywhere.